server/container/guice/mailbox-postgres/src/main/java/org/apache/james/modules/mailbox/PostgresMailboxModule.java

```diff
@@ -133,6 +135,7 @@ protected void configure() {
         bind(AttachmentMapperFactory.class).to(PostgresMailboxSessionMapperFactory.class);
 
         bind(ReIndexer.class).to(ReIndexerImpl.class);
+        bind(MessageIdReIndexer.class).to(MessageIdReIndexerImpl.class);
```

Comment:
What was the issue when we were missing this? Just curious :)

Reply:
It is required for web admin MessageRoute

Reply:
Wait wait
Do we need MessasgeRoute for postgres?
https://james.apache.org/server/manage-webadmin.html#Administrating_Messages
```
This task is only available on top of Guice Cassandra products or Guice JPA products. It is not part of Memory Guice product.
```
This is an old document, not sure it still correct or not with Postgres

Reply:
likely we need e.g. reindex messages in OpenSearch?

Reply:
As postgres can use Opensearch fo idexing, yes I think we need the task for reindexing :)